src/stackdriver-nozzle/main.go

```diff
@@ -48,21 +54,32 @@ func main() {
 	//todo: pull in logging library...
 	kingpin.Parse()
 
-	client := firehose.NewClient(*apiEndpoint, *username, *password, *skipSSLValidation)
+	input := firehose.NewClient(*apiEndpoint, *username, *password, *skipSSLValidation)
 
+	var output firehose.FirehoseHandler
 	var err error
+
 	if *debug {
 		println("Sending firehose to standard out")
-		err = client.StartListening(&dev.StdOut{})
+		output = &dev.StdOut{}
 	} else {
 		println("Sending firehose to Stackdriver")
 		sdClient := stackdriver.NewClient(*projectID, *batchCount, *batchDuration)
-		n := nozzle.Nozzle{StackdriverClient: sdClient}
-
-		err = client.StartListening(&n)
+		output = &nozzle.Nozzle{StackdriverClient: sdClient}
 	}
-	if err != nil {
+	if err != nil || output == nil {
 		panic(err)
 	}
 
+	filteredOutput, err := filter.New(output, strings.Split(*eventsFilter, ","))
+
+	if err != nil {
+		println("Error:", err.Error())
+		filter.DisplayValidEvents()
+		return
```

Comment:
https://golang.org/pkg/os/#Exit with non-0 might be a better choice than returning